File: 383. Ransom Note/Java/Ransom Note.java
public class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        Map<Character, Integer> has = new HashMap<>();
        for (char item : magazine.toCharArray()) {
            if (!has.containsKey(item)) {
                has.put(item, 0);
            }
            has.put(item, has.get(item) + 1);
        }
        for (char item : ransomNote.toCharArray()) {
            if (!has.containsKey(item) || has.get(item) == 0) {
                return false;
            }
            has.put(item, has.get(item) - 1);
        }
        return true;
    }
}

File: 383. Ransom Note/Python/Ransom Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        has = {}
        for item in magazine:
            if item not in has:
                has[item] = 0
            has[item] += 1
        for item in ransomNote:
            if item not in has or has[item] == 0:
                return False
            has[item] -= 1
        return True
